docs/adr/0038-use-esbuild-in-production.md

```diff
@@ -0,0 +1,63 @@
+# Use esbuild for production builds
+
+- Status: TBD
+- Deciders: Ben Bangert, Wil Clouser, Dan Schomburg, Barry Chen, Julian Poyourow, Lauren Zugai, Vijay Budhram
+- Date: 2023-03-29
+
+## Context and Problem Statement
+
+We currently use esbuild throughout the code base for running tests and starting up local development servers.
+This has proven to be much faster and better for developer experience than compiling with typescript.
+
+We do not, however, use esbuild in production. This means we compile our code using tsc just prior to deployment.
+This is slow when compared to using esbuild, and often results in last minute typescript errors being encountered
+on deployment.
+
+## Decision Drivers
+
+- Faster deployments
+- Better parity between test/development and production
+- Ability to retain type checking
+- Ability to catch typescript errors early
+
+## Considered Options
+
+- A: Switch to using esbuild in production and add CI job for type checking
+- B: Continue using esbuild for dev/test and tsc for production
+- C: Switch everything over to tsc
+
+## Decision Outcome
+
+After a decent amount of discussion, we have decided to switch over to esbuild for our production builds. Other
+teams have made the switch and not encountered major issues.
```

Comment:
What other teams is this referring to?

Reply:
Not sure about plural, but I know the [Firefox Monitor team](https://github.com/mozilla/blurts-server/blob/main/src/esbuild.js) switched to esbuild for their big v2 redesign. Amri did a good demo of their new tech stack in a recent 5x8 presentation a couple weeks ago (that I can probably dig up on AirMo if you are interested).

Reply:
Maybe the plural is a misleading. I'll qualify this. And yes, I was referring to the Firefox monitor team.

Reply:
There are others, but we definitely seem to be pioneers: https://github.com/search?q=esbuild+user%3Amozilla+path%3A**%2Fpackage.json+lang%3AJSON&type=code

@clouserw might also have more context since I think he was speaking w/ the Monitor team and possibly others.

Reply:
I was talking with Amri: https://mozilla.slack.com/archives/CEV7MGSQH/p1678894690315309